lib/index.js

```diff
@@ -290,23 +435,74 @@ module.exports = class OtpRunner {
       await this.fail('Build graph failed! Please see logs.')
     }
     this.status.graphBuilt = true
+
+    await this.updateStatus(
+      'Graph built successfully!',
+      this.manifest.runServer ? 70 : 90
+    )
+
+    // add various follow-up tasks that can occur independently of the server
+    // startup
+    this.postBuildTasks.push(this.uploadGraphObj())
+    this.postBuildTasks.push(this.uploadBuildLogs())
+    this.postBuildTasks.push(this.createAndUploadBundle())
+    this.postBuildTasks.push(this.uploadGraphBuildReport())
```

Comment:
Do we want these to be independent? What happens if one fails? At the very least, I think the comment should let the reader know what happens in that case.

Reply:
I believe there is an advantage to having them be asynchronous so that other processes like starting the server don't have to wait for the graph to be uploaded. If the build logs or graph build report fail to be uploaded, the overall job won't be failed. The bundle task is an unimplemented stub. If the uploadGraphObj fails, that will cause an overall failure. So it seems the worst thing that could happen is that if the graph upload fails, then make the graph build logs and report might not have their result logged in time before the whole program stops. I added a comment in commit 2610ee9df6cb1878f067a5e2c1c277eed33836fe to the `uploadFileToS3` method to note that a failure to upload to s3 merely returns false. I believe further discussion of how to handle these failure cases is out of the scope of this PR.

Reply:
I'm not even sure what `createAndUploadBundle` is a stub for. Can you expand on that in a TODO comment here?

Also, after looking at this a bit more, there's a lot of redirection happening with the `postBuildTasks`. This could probably be helped by defining the postBuildTasks a little closer to where they are all collected:

```javascript
this.postBuildTasks = []
if (this.manifest.runServer) {
  // Make running the server the very first post build task
  this.postBuildTasks.push(this.startServer())
}
// Add various follow-up tasks that can occur independently of the server
// startup
this.postBuildTasks.push(
  this.uploadGraphObj(),
  this.uploadBuildLogs(),
  this.createAndUploadBundle(),
  this.uploadGraphBuildReport()
)
```

Reply:
Refactored in 7a9a956a6b31abb6b6607ed8ec8f73fef1d0fd76